src/FubuMVC.Core/ServiceBus/Configuration/ChannelGraph.cs

```diff
@@ -112,6 +111,20 @@ public Uri ReplyChannelFor(string protocol)
             return _replyChannels[protocol];
         }
 
+        public List<string> GetAcceptedContentTypesForChannel(Uri channelUri)
+        {
+            var acceptedContentTypes = new List<string>();
+            var node = this.FirstOrDefault(x => x.Channel.Address == channelUri) ?? this.FirstOrDefault(x => x.Uri == channelUri);
+
+            acceptedContentTypes.AddRange(node?.AcceptedContentTypes ?? Enumerable.Empty<string>());
+            acceptedContentTypes.Add(node?.DefaultContentType);
+            acceptedContentTypes.Add(node?.DefaultSerializer?.ContentType);
+            acceptedContentTypes.AddRange(AcceptedContentTypes);
+            acceptedContentTypes.Add(DefaultContentType);
+
+            return acceptedContentTypes.Where(x => x != null).Distinct().ToList();
```

Comment:
The Distinct() might end up losing the ordering. Is that what you meant to do here? Wouldn't we want to rank preferred mimetypes to make sure it can opt into the most efficient serializer?